### PR TITLE
More dogfooding of `string split -f`

### PR DESCRIPTION
**Pull request by ammgws**
Continuing on from #6874

### CODE REVIEW

File: share/functions/__fish_crux_packages.fish [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # a function to obtain a list of installed packages with CRUX pkgutils
 function __fish_crux_packages -d 'Obtain a list of installed packages'
-    pkginfo -i | cut -d' ' -f1
+    pkginfo -i | while read -l line
```
Comment (krobelus) [inline share/functions/__fish_crux_packages.fish:3]: The `cut` version was quite short and sweet. Isn't it possible to do `string split -f1 ' ' -- (pkginfo -i)`?

Comment (ammgws) [reply share/functions/__fish_crux_packages.fish:3]: Due to how `string split` functions the fish version gets a bit longer, but hopefully it isn't an issue if the aim of fish is not to be as short as possible.

```sh
>printf "package1 description1\npackage2 description2\n" | cut -d' ' -f1
package1
package2

>printf "package1 description1\npackage2 description2\n" | string split -f1 ' '
package1

>printf "package1 description1\npackage2 description2\n" | while read -l line
    string split -f1 ' ' -- $line
 end
package1
package2
```

Comment (krobelus) [reply share/functions/__fish_crux_packages.fish:3]: I think it's surprising that `string split` only accepts one argument:
I would expect `string split -f1 ' ' 'a b' 'c d'` to print `a` and `c`.

Comment (ammgws) [reply share/functions/__fish_crux_packages.fish:3]: Yeah I came across that which is why I ended up going with the `read`.

**Comment by ammgws** [comment]
Addressed with #6911

**Pull request closed without merging**